File: hackerrank/minimumSwaps.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {


static void swap(int[] array,int left, int right){
        int temp = array[right];
        array[right] = array[left];
        array[left] = temp;
    }
        
static int minimumSwaps(int[] arr) {
        int rightPointer = arr.length -1;

        int count = 0;
        int minimumSwaps = 0;

        while(count < arr.length){
            int arrValue = count+1;

            if(arr[count] == arrValue){
                count++;
                continue;
            }

            while(arr[rightPointer] != arrValue){
                rightPointer --; // finds the actual position of current index value.  
            }

            if(rightPointer != count){ // actual position and current index is different, swap
                swap(arr, count, rightPointer);
                minimumSwaps++;
            }

            rightPointer = arr.length -1;
            count++;
        }
        return minimumSwaps;
    }


    // Complete the minimumSwaps function below.
   // static int minimumSwaps(int[] arr) {


    //}

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] arr = new int[n];

        String[] arrItems = scanner.nextLine().split(" ");
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int i = 0; i < n; i++) {
            int arrItem = Integer.parseInt(arrItems[i]);
            arr[i] = arrItem;
        }

        int res = minimumSwaps(arr);

        bufferedWriter.write(String.valueOf(res));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}

File: hackerrank/minimumSwaps.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the minimumSwaps function below.
def minimumSwaps(arr):
    '''
    i = 1
    #first = a[0]
    length = len(arr)
    swaps = 0
    for i in range(length):
        for j in range(length):
            if arr[i] < arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
                swaps = swaps + 1
    print(arr)
    return swaps
    '''
    ## The problem clearly says, the numbers are consecutive and unordered.
    ## Logic: Check if the number is in it's designated index.
    ## 1 should be at the position 0, 2 at 1, 3 at 2, etc.,
    ## if not, find the position of the number and swap it. Do for all elements. 
    counter = 0
    for i in range(len(arr)):
        if (arr[i]) != i+1:
                l = arr.index(i+1)
                print(l)
                arr[i],arr[l] = i+1,arr[i]
                counter += 1
        print(arr)
    return counter

print(minimumSwaps([7, 1, 3, 2, 4, 5, 6]))

'''
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    res = minimumSwaps(arr)

    fptr.write(str(res) + '\n')

    fptr.close()
'''